lilaclib.py

```diff
@@ -111,20 +117,41 @@ def download_official_pkgbuild(name):
       f.write(data)
   return files
 
+def try_aur_url(name):
+  aur4mig = 'https://aur4.archlinux.org/cgit/aur.git/snapshot/{name}.tar.gz'
+  aur3url = 'https://aur.archlinux.org/packages/{first_two}/{name}/{name}.tar.gz'
+  aur4url = 'https://aur.archlinux.org/cgit/aur.git/snapshot/{name}.tar.gz'
+  templates = [aur4mig, aur3url, aur4url]  # TODO: reorder this after migration to aur4
+  urls = [url.format(first_two=name[:2], name=name) for url in templates]
+  for url in urls:
+    response = requests.get(url)
+    if response.status_code == 200:
+      logger.debug("downloaded aur tarball '%s' from url '%s'" % (name, url))
+      return response.content
+  logger.error("failed to find aur url for '%s'" % name)
+  raise AurDownloadError(name)
+
 def download_aur_pkgbuild(name):
-  url = 'https://aur.archlinux.org/packages/{first_two}/{name}/{name}.tar.gz'
-  url = url.format(first_two=name[:2], name=name)
-  lilac_aur = "lilac_aur"
-  run_cmd(['sh', '-c', "curl '%s' | tar xzv --one-top-level='%s' --strip-components=1" % (url, lilac_aur)])
-  for f in ('.AURINFO', '.SRCINFO'):
-    try:
-      os.unlink(os.path.join(lilac_aur, f))
-    except FileNotFoundError:
-      pass
-  files = os.listdir(lilac_aur)
-  for f in files:
-    os.rename(os.path.join(lilac_aur, f), f)
-  os.rmdir(lilac_aur)
+  tar_content = BytesIO(try_aur_url(name))
+  tar_file = None
+  files = []
+  try:
+    tar_file = tarfile.open(name=name+".tar.gz", mode="r:gz", fileobj=tar_content)
+    for filename in tar_file.getnames():
+      basename, remain = os.path.split(filename)
+      if basename == '':
+        continue
+      if remain in ('.AURINFO', '.SRCINFO'):
+        continue
+      contentfile = open(remain, "wb")
+      contentfile.write(tar_file.extractfile(filename).read())
+      contentfile.close()
+      files.append(remain)
+  finally:
+    if tar_file:
+      tar_file.close()
+    if tar_content:
+      tar_content.close()
```

Comment:
No need to close a `BytesIO`. Also tarfiles support `with` so you don't need to try.